python/semantic_kernel/connectors/memory/qdrant/qdrant_collection.py

```diff
@@ -0,0 +1,257 @@
+else:
+    from typing_extensions import override  # pragma: no cover
+
+from pydantic import ValidationError
+from qdrant_client.async_qdrant_client import AsyncQdrantClient
+from qdrant_client.models import PointStruct, VectorParams
+
+from semantic_kernel.connectors.memory.qdrant.const import DISTANCE_FUNCTION_MAP, TYPE_MAPPER_VECTOR
+from semantic_kernel.connectors.memory.qdrant.utils import AsyncQdrantClientWrapper
+from semantic_kernel.data.vector_store_model_definition import VectorStoreRecordDefinition
+from semantic_kernel.data.vector_store_record_collection import VectorStoreRecordCollection
+from semantic_kernel.data.vector_store_record_fields import VectorStoreRecordVectorField
+from semantic_kernel.exceptions import (
+    MemoryConnectorInitializationError,
+    VectorStoreModelValidationError,
+)
+from semantic_kernel.exceptions.memory_connector_exceptions import MemoryConnectorException
+from semantic_kernel.kernel_types import OneOrMany
+from semantic_kernel.utils.experimental_decorator import experimental_class
+from semantic_kernel.utils.telemetry.user_agent import APP_INFO, prepend_semantic_kernel_to_user_agent
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+TModel = TypeVar("TModel")
+TKey = TypeVar("TKey", str, int)
+
+
+@experimental_class
+class QdrantCollection(VectorStoreRecordCollection[str | int, TModel]):
+    """A QdrantCollection is a memory collection that uses Qdrant as the backend."""
+
+    qdrant_client: AsyncQdrantClient
+    named_vectors: bool
+    supported_key_types: ClassVar[list[str] | None] = ["str", "int"]
+    supported_vector_types: ClassVar[list[str] | None] = ["float", "int"]
+
+    def __init__(
+        self,
+        data_model_type: type[TModel],
+        data_model_definition: VectorStoreRecordDefinition | None = None,
+        collection_name: str | None = None,
+        named_vectors: bool = True,
+        url: str | None = None,
+        api_key: str | None = None,
+        host: str | None = None,
+        port: int | None = None,
+        grpc_port: int | None = None,
+        path: str | None = None,
+        location: str | None = None,
+        prefer_grpc: bool | None = None,
+        client: AsyncQdrantClient | None = None,
+        env_file_path: str | None = None,
+        env_file_encoding: str | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Initializes a new instance of the QdrantVectorRecordStore.
+
+        When using qdrant client, make sure to supply url and api_key.
+        When using qdrant server, make sure to supply url or host and optionally port.
+        When using qdrant local, either supply path to use a persisted qdrant instance
+            or set location to ":memory:" to use an in-memory qdrant instance.
+        When nothing is supplied, it defaults to an in-memory qdrant instance.
+        You can also supply a async qdrant client directly.
+
+        Args:
+            data_model_type (type[TModel]): The type of the data model.
```

Comment:
For doc generation, isn't it better to not include the type like (type[TModel])?

Reply:
yeah, but that needs to be done across the board, I'm looking at that separately.